### DISCUSSION TITLE
Casting oblivious nullable type to nullable and dereferencing does not report warning

### DISCUSSION THREAD

**Discussion by duracellko** [dotnet/csharplang#2575] [General] [answered]
When a method returns oblivious nullable type and it is assigned to explicitly nullable variable, compiler still does not report warning, when dereferencing the variable.

More details in sample repository [https://github.com/duracellko/CS8Nullable](https://github.com/duracellko/CS8Nullable).

**Comment by YairHalberstadt** [comment, marked as answer]
This is by design. The type a variable is assigned to does not affect its nullable flow state.

**Comment by huoyaoyuan** [comment]
https://sharplab.io/#v2:D4AQTAjAsAUCDMACciDCiDetE8QYgDsBXAGxIEMAjEgU0QBMBLAZytu1045wWQgAZEAZQAUASkQBeAHyJiZANzdEywqQrU6NAmxrLeIACyIAsuOVYYnTiAEB+RMymIRt/nbGixSq9ZzMAOgBxGgAXAAlyZgALVAB7ehpxH04AX1hUoA=

If you do an explicit cast, then there is a warning reported.
